File: llm_generated_code/ChatGPT/CPP/Level2/Q16.cpp
#include <iostream>
#include <vector>
using namespace std;

int findMissingTreasure(int N, const vector<int>& treasures) {
    int expectedSum = N * (N + 1) / 2;
    int actualSum = 0;
    for (int treasure : treasures) {
        actualSum += treasure;
    }
    return expectedSum - actualSum;
}

int main() {
    int N = 5;
    vector<int> treasures = {1, 2, 3, 5};
    cout << "Missing treasure: " << findMissingTreasure(N, treasures) << endl;
    return 0;
}

File: llm_generated_code/ChatGPT/Python/Level2/Q16.py
def find_missing_treasure(treasures, N):
    total_sum = N * (N + 1) // 2
    found_sum = sum(treasures)
    return total_sum - found_sum

treasures = [1, 2, 3, 5]
N = 5
print(find_missing_treasure(treasures, N))
